docs/contributing.md

```diff
@@ -0,0 +1,172 @@
+  - Temporary backports or in-flight patches are acceptable if tracked.
+  References:
+  - [Yocto kernel workflow](https://docs.yoctoproject.org/kernel-dev/index.html)
+  - [Submitting Linux patches](https://www.kernel.org/doc/html/latest/process/submitting-patches.html)
+- **Firmware:**
+  Custom firmware must be contributed to [`linux-firmware`](https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/) whenever possible.
+- **External layer recipes:**
+  Changes targeting `oe-core` or `meta-openembedded` should be sent directly upstream.
+
+---
+
+## 4  Downstream Baseline – Qualcomm Linux 1.x
+
+**Goal:** extend existing Qualcomm Linux 1.x releases with third-party hardware support equivalent to official EVKs.
+**Target branch:** `scarthgap` (≥ 1.4)
+
+### 4.1  Expected Contribution Types
+
+- Machine configuration files
+- Recipes for closed-source components
+  (same binary-hosting rules as upstream)
+- Machine-specific packagegroups:
+  [`meta-qcom-hwe/recipes-firmware/packagegroups`](https://github.com/qualcomm-linux/meta-qcom-hwe/tree/scarthgap/recipes-firmware/packagegroups)
+- Partition definitions:
+  [`meta-qcom-hwe/recipes-devtools/partition-utils`](https://github.com/qualcomm-linux/meta-qcom-hwe/tree/scarthgap/recipes-devtools/partition-utils)
+- Kernel customization:
+  Follow [`linux-qcom-base`](https://github.com/qualcomm-linux/meta-qcom-hwe/tree/scarthgap/recipes-kernel/linux)
+  or `linux-qcom-custom` guidelines.
+  - Machine-specific patches via `.bbappend`.
+  - Custom kernel recipe forks only with maintainer consent.
+- Firmware additions:
+  Extend [`linux-firmware` recipes](https://github.com/qualcomm-linux/meta-qcom-hwe/tree/scarthgap/recipes-firmware/firmware) as needed.
+- Hardware-specific enablement should **live entirely within this layer**, not in external “extras” layers or manifests.
+- Vendor-specific distro features / demo content belong in a **separate distro layer** maintained by the vendor.
+
+---
+
+## 5  Integration Expectations
+
+- The repository should maintain **CI / CD** pipelines (GitHub Actions with kas) to validate new submissions.
+  Reference: [kas Documentation](https://kas.readthedocs.io/en/latest/userguide.html)
+- Vendors may optionally integrate with **LAVA** for runtime testing (get in contact with the project maintainers or raise an issue).
+  Reference: [LAVA Framework](https://docs.lavasoftware.org/lava/index.html)
+
+---
+
+## 6  Machine Example – Arduino Uno-Q
+
+*TODO*
+
+The `Arduino Uno-Q` board serves as an example of a machine integration inside this layer.
+
+Typical components include:
+- `conf/machine/arduino-unoq.conf`
+- Board-specific packagegroup under `recipes-bsp/packagegroups/`
+- Firmware recipes referencing vendor-hosted binaries
+- Partition configuration defined through `qcom-ptool`
+- Kernel support via `linux-qcom-next` `.bbappend`
+- CI job validating `core-image-minimal` build and boot on target
```

Comment:
I understand `core-image-minimal` does not include non-vital firmware, which can be binaries hosted on an vendor mirror. It means (I believe) that building only `core-image-minimal` will not detect missing closed firmware files. If this is the intended, so ok

Reply:
Right, we do test core-image-base actually (will update), but modules are not included by default there, which is ok as we also have the base module list available in the bsp layer.